barretenberg/cpp/src/barretenberg/commitment_schemes_recursion/ipa_recursive.test.cpp

```diff
@@ -60,23 +61,78 @@ class IPARecursiveTests : public CommitmentTest<NativeCurve> {
         return { recursive_verifier_transcript, stdlib_opening_claim };
     }
 
+    Builder build_ipa_recursive_verifier_circuit(const size_t POLY_LENGTH)
+    {
+        Builder builder;
+        auto recursive_verifier_ck = std::make_shared<VerifierCommitmentKey<Curve>>(&builder, POLY_LENGTH, this->vk());
+        auto [stdlib_transcript, stdlib_claim] = create_ipa_claim(builder, POLY_LENGTH);
+
+        RecursiveIPA::reduce_verify(recursive_verifier_ck, stdlib_claim, stdlib_transcript);
+        builder.finalize_circuit(/*ensure_nonzero=*/true);
+        return builder;
+    }
+
     /**
      * @brief Tests IPA recursion
      * @details Creates an IPA claim and then runs the recursive IPA verification and checks that the circuit is valid.
      * @param POLY_LENGTH
      */
     void test_recursive_ipa(const size_t POLY_LENGTH)
     {
-        Builder builder;
-        auto recursive_verifier_ck = std::make_shared<VerifierCommitmentKey<Curve>>(&builder, POLY_LENGTH, this->vk());
-        auto [stdlib_transcript, stdlib_claim] = create_ipa_claim(builder, POLY_LENGTH);
-
-        RecursiveIPA::reduce_verify(recursive_verifier_ck, stdlib_claim, stdlib_transcript);
-        builder.finalize_circuit(/*ensure_nonzero=*/false);
+        Builder builder(build_ipa_recursive_verifier_circuit(POLY_LENGTH));
         info("IPA Recursive Verifier num finalized gates = ", builder.get_num_finalized_gates());
         EXPECT_TRUE(CircuitChecker::check(builder));
     }
 
+    void test_fixed_ipa_recursive_verifier()
+    {
+
+        srs::init_crs_factory("../srs_db/ignition");
+
+        Builder builder_1(build_ipa_recursive_verifier_circuit(1 << 10));
+        Builder builder_2(build_ipa_recursive_verifier_circuit(1 << 11));
+
+        UltraFlavor::ProvingKey pk_1 = (DeciderProvingKey_<UltraFlavor>(builder_1)).proving_key;
+        UltraFlavor::ProvingKey pk_2 = (DeciderProvingKey_<UltraFlavor>(builder_2)).proving_key;
+        UltraFlavor::VerificationKey verification_key_1(pk_1);
+        UltraFlavor::VerificationKey verification_key_2(pk_2);
+        bool broke(false);
+        auto check_eq = [&broke](auto& p1, auto& p2) {
```

Comment:
I suppose its useful to have all of this logic here for debugging purposes but just FYI I believe the operators are in place so you can also just check equality of verification keys directly (just sometimes have to manually exclude the commitment keys depending on how they were constructed)

Reply:
ah ok, yeah I just wanted to have the debugging logic